sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala

```diff
@@ -366,6 +356,23 @@ trait Interpreter extends ScorexLogging {
     fullJitCost
   }
 
+  /** Returns evaluation mode used by this interpreter in the given context.
+    * By default evaluation mode is determined based on `context.activatedScriptVersion`
+    * so that the interpreter works either as v4.x of v5.0.
+    *
+    * Alternatively, the required evaluation mode can be specified by giving Some(mode)
+    * value in `evalSettings.evaluationMode`, in which case the interpreter works as
+    * spedified.
+    */
+  def getEvaluationMode(context: CTX): EvaluationMode = {
+    evalSettings.evaluationMode.getOrElse {
+      if (context.activatedScriptVersion < Interpreter.MaxSupportedScriptVersion)
```

Comment:
It is better  to introduce a new constant like JitActivationVersion, then there is no need to check MaxSupportedScriptVersion usages before increasing it in future (which can be easily missed)

Reply:
done